internal/mode/static/handler.go

```diff
@@ -101,13 +101,13 @@ func (h *eventHandlerImpl) HandleEventBatch(ctx context.Context, batch events.Ev
 }
 
 func (h *eventHandlerImpl) updateNginx(ctx context.Context, conf dataplane.Configuration) error {
-	files := h.cfg.generator.Generate(conf)
+	files, configVersion := h.cfg.generator.Generate(conf)
```

Comment:
I think this might be cleaner because we wouldn't have to modify the Generator interface to return an integer

Reply:
Sorry, I missed this! Yeah, that makes sense, I'll change it

Reply:
Sorry, I missed this! I'll take a look

Reply:
Ok I pushed a commit moving the version to the `dataplane.Configuration` type, but I'm not sure about the approach because now the Generate function is mutating the conf and it doesn't feel quite right. Ideally, it would be added above when we are generating the conf instead of when we are generating the files, but then I'm not sure where we could store this version state. What do you think?

Reply:
> Ideally, it would be added above when we are generating the conf instead of when we are generating the files, but then I'm not sure where we could store this version state.

What if the event handler kept this state and passed the version to the BuildConfiguration function?